matrix_sdk/src/request_builder.rs

```diff
@@ -310,7 +310,7 @@ impl RegistrationBuilder {
     /// If this does not correspond to a known client device, a new device will be created.
     /// The server will auto-generate a device_id if this is not specified.
     pub fn device_id<S: Into<String>>(&mut self, device_id: S) -> &mut Self {
```

Comment:
This should probably just be `Into<Box<DeviceId>>`, then the body doesn't need to change. I was thinking I might create a separate PR only bumping the Ruma version that, so this one would then be rebased after merging the first. Maybe you should still extract the commits doing unrelated porting work into a separate PR. @poljar's call.

Reply:
I went with `Into<Box<str>>` after splitting the commits up